GitUI/CommandsDialogs/RepoHosting/CreatePullRequestForm.cs

```diff
@@ -104,11 +106,17 @@ private void _pullReqTargetsCB_SelectedIndexChanged(object sender, EventArgs e)
             _remoteBranchesCB.Items.Clear();
             _remoteBranchesCB.Text = _strLoading.Text;
 
-            AsyncLoader.DoAsync(
-                () => _currentHostedRemote.GetHostedRepository().Branches,
-                branches =>
+            // TODO merge with nearly identical code below
+            ThreadHelper.JoinableTaskFactory.RunAsync(
+                async () =>
                 {
-                    branches.Sort((a, b) => string.Compare(a.Name, b.Name, true));
+                    await TaskScheduler.Default;
+
+                    var branches = _currentHostedRemote.GetHostedRepository().Branches;
+                    branches.Sort((a, b) => string.Compare(a.Name, b.Name, StringComparison.OrdinalIgnoreCase));
+
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
💡  This should use the extension method form to ensure the UI is not disposed

Reply:
Thanks. Fixed this and other instances.